### DISCUSSION TITLE
Function with Event GUILD_MEMBER_ADD

### DISCUSSION THREAD

**Discussion by DevExploris** [discord-php/DiscordPHP#1179] [Q&A] [answered]
Hello,

It's my first intervention in github. If any information is missing, please let me know.

I want to create a DM  when a user join the guild (it's a training).

This is my code :

$discord->on(Event::GUILD_MEMBER_ADD, function (Member $member, Discord $discord) {
    
    $newMember = $member->user;

    $newMember->sendMessage("something");

});

I've been through a lot of issues about the GUILD_MEMBER_ADD but it still doesn't work for me.

On my developper panel, I checked the permission "SERVER_MEMBER_INTENT" and my bot is administrator of the server.

When I watch the service log when user join the guild, nothing at all.

I'have testing with Event::message and it's working.

Can you help me about this problem ?

**Comment by key2peace** [comment, marked as answer]
have you done this as well?
```php
use Discord\Discord;
use Discord\WebSockets\Intents;

$discord = new Discord([
	'token' => 'xxx',
	'intents' => Intents::getDefaultIntents() | Intents::GUILD_MEMBERS,
]);
```

**Reply by DevExploris** [reply]
Hello,

Thank you for your help.

I added Intents:GUILD_MEMBERS and it's worked !

Thank you very much !
